src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionSerializer.cs

```diff
@@ -87,5 +88,73 @@ internal void WriteCollectionEnd()
                 this.JsonWriter.EndObjectScope();
             }
         }
+
+        /// <summary>
+        /// Asynchronously writes the start of a collection.
+        /// </summary>
+        /// <param name="collectionStart">The collection start to write.</param>
+        /// <param name="itemTypeReference">The item type of the collection or null if no metadata is available.</param>
+        internal async Task WriteCollectionStartAsync(ODataCollectionStart collectionStart, IEdmTypeReference itemTypeReference)
+        {
+            Debug.Assert(collectionStart != null, "collectionStart != null");
+
+            if (this.writingTopLevelCollection)
+            {
+                // "{"
+                await this.AsynchronousJsonWriter.StartObjectScopeAsync()
+                    .ConfigureAwait(false);
+
+                // "@odata.context":...
+                await this.WriteContextUriPropertyAsync(
+                    ODataPayloadKind.Collection,
+                    () => ODataContextUrlInfo.Create(collectionStart.SerializationInfo, itemTypeReference))
+                        .ConfigureAwait(false);
+
+                // "@odata.count":...
+                if (collectionStart.Count.HasValue)
+                {
+                    await this.AsynchronousODataAnnotationWriter.WriteInstanceAnnotationNameAsync(ODataAnnotationNames.ODataCount)
+                        .ConfigureAwait(false);
+                    await this.AsynchronousJsonWriter.WriteValueAsync(collectionStart.Count.Value)
+                        .ConfigureAwait(false);
+                }
+
+                // "@odata.nextlink":...
+                if (collectionStart.NextPageLink != null)
+                {
+                    await this.AsynchronousODataAnnotationWriter.WriteInstanceAnnotationNameAsync(ODataAnnotationNames.ODataNextLink)
+                        .ConfigureAwait(false);
+                    await this.AsynchronousJsonWriter.WriteValueAsync(this.UriToString(collectionStart.NextPageLink))

```

Comment:
Maybe `AsyncJsonWriter` :)?

Reply:
Will take this into consideration but it'd have to be a PR by itself since the naming was adopted in an already merged PR and it'd involve changing in quite a few places.